### ISSUE TITLE
Add a recommendation to use .shouldComponentUpdate or React.PureComponent

### ISSUE THREAD

**Issue by prometheansacrifice** [garbles/why-did-you-update#35]
As opposed to just show that props are unchanged.

**Comment by caesarsol** [comment]
I wouldn't, because there are many better ways to prevent wasted renders, especially if using Redux or Mobx. And there could be new methods in the future.

**Comment by prometheansacrifice** [comment]
Better ways? Can you elaborate?

**Comment by bf** [comment]
Yes please elaborate. The readme should at least point to some resources and showcase different the approaches.

**Comment by saitonakamura** [comment]
I think the point of @caesarsol is that avoiding re-render is not only about `PureComponent` and implementing `shoudComponentUpdate` by hand, but a much more wider topic. For example, if you're using `connect` from `react-redux` to pass state parts to props, it has `pure` option which is `true` by default and does the same thing as `PureComponent`. Or if you're using stateless component, you can't implement `shouldComponentUpdate`, but instead of converting to `PureComponent` you can use [`pure` function from `recompose`](https://github.com/acdlite/recompose#optimize-rendering-performance)
So advocating using `PureComponent` or `shouldComponentUpdate` looks slighly opinionated

**Comment by caesarsol** [comment]
Sorry for the delay.

That's exactly my point!

For example the recommended way to better performance in my opinion is to divide in multiple connected components, which render independently.

But it's not an absolite rule because too many connected components can actually slow down the page...

The important thing is that `shouldComponentUpdate` is mostly discouraged, because is a step back to imperative programming while React tends to be declarative.

Redux has `connect` which already optimizes renders, if you use it correctly.

I'm sure @gaearon could give many more advices, and already has somewhere between Medium and Twitter :)
